modules/oauth/src/test/java/org/mule/module/oauth2/internal/clientcredentials/functional/ClientCredentialsFailureTestCase.java

```diff
@@ -67,8 +67,15 @@ public void run() throws Exception
                 ApplicationContextBuilder applicationContextBuilder = new ApplicationContextBuilder().setApplicationResources(new String[] {"client-credentials/client-credentials-minimal-config.xml"});
                 MuleContext muleContext = applicationContextBuilder.build();
                 
-                expectedException.expectCause(isA(IOException.class));
-                ((Flow)(muleContext.getRegistry().lookupFlowConstruct("testFlow"))).process(MuleTestUtils.getTestEvent("", muleContext));
+                try
+                {
+                    expectedException.expectCause(isA(IOException.class));
+                    ((Flow)(muleContext.getRegistry().lookupFlowConstruct("testFlow"))).process(MuleTestUtils.getTestEvent("", muleContext));
+                }
+                finally
+                {
+                    muleContext.dispose();
```

Comment:
Would this have caused a failure without the fix?

Reply:
no, the issue is just manifested in the list of the dangling threads after each test